timelines/base_timeline.py

```diff
@@ -0,0 +1,14 @@
+class BaseTimeline:
+    def __init__(self):
+        self.events = []
+        self.instance = 0
+        print("Initalizing BaseTimeline")
+
+    def register_event(self, event):
+        print(f'Registering event : {type(event).__name__}')
+        self.events.append(event)
+ 
+    def step(self, state):
+        for event in self.events:
+            event.step(self.instance, state)
+        self.instance += 1
```

Comment:
So this part I’m confused on, I know the step part isn’t fleshed out yet being only a print statement, but wouldn’t this for-loop realistically due to the nomenclature step through this event.step for as many times as there are events iterating several times on the same instance?

Reply:
I'm not entirely sure what you mean - but I think your above comment about making per-person will help fix this up some and how events are implemented, so I'll wait on that before merging so I can clean up the rest.

I'm assuming your question is in regards to populations?  I wasn't sure the way this made the most sense - so if you have thoughts I think either way would work - and you've thought through big picture more too.

1.) Timeline > Events > Run per population
ie.) Generic 'mid-life crisis' event that effects X population
2.) Timeline > Population > Run events that are registered per individual
ie.) Specific or generated event that is specifically assigned to 'Johnny'

If that above example makes sense?

I think we're thinking of events differently - cause I know you had in mind language parsing stuff which would be I think more per individuals rather than generic events that effect populations
.
I was thinking that you could define a generic event as a class ie.) 'Mid-life crisis event' that will hit populations with X level of Y trait and have Z effect.
Events would basically have a query function they also define that determines which set of the population to effect and when.
The state that gets passed in would be a set of individuals that it can query through.

Also conceptually idk if that makes sense to do at the event level.

🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️ 🤷‍♂️